lib/widgets/reusable_widgets/dialogs.dart

```diff
@@ -57,25 +57,29 @@ showWarningDialog({
 showDialogWithNoAndYesOptions({
   required BuildContext context,
   required String title,
-  required String description,
   required VoidCallback onYesButtonPressed,
+  Widget? content,
+  String? description,
+  VoidCallback? onNoButtonPressed,
 }) =>
     showDialog(
+      barrierDismissible: false,
       context: context,
       builder: (context) => AlertDialog(
         title: Text(title),
-        content: Text(description),
+        content: content ?? Text(description!),
         actions: [
           TextButton(
             child: Text(
               'No',
               style: Theme.of(context).textTheme.bodyLarge,
             ),
             onPressed: () {
-              Navigator.of(context).pop();
+              onNoButtonPressed?.call();
             },
           ),
           TextButton(
+            // TODO: add also pop call
```

Comment:
There's an issue here. It could just be that pop is missing, but I'm unable to decline a token transfer once this dialog is displayed.
I should be able to decline by pressing the `No` option or clicking outside the dialog.